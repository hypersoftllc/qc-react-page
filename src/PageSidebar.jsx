import PT from 'prop-types'
import React from 'react'
import Block from '@qc/react-block'


function PageSidebar(props) {
  const domProps = {
    ...props,
    className: `${props.compClassName} ${props.className}`.trim(),
  }
  delete domProps.children
  delete domProps.compClassName
  return (
    <Block {...domProps}>
      {props.children}
    </Block>
  )
}

PageSidebar.defaultProps = {
  className: '',
  compClassName: 'Page-Sidebar',
  compType: 'aside',
}

PageSidebar.propTypes = {
  children: PT.node,
  className: PT.string,
  compClassName: PT.string,
  compType: PT.oneOf(['aside', 'div', 'section']),
}

export default PageSidebar
