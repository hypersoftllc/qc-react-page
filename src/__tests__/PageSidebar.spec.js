/* eslint-env jest */
/* eslint padded-blocks: "off" */

import React from 'react'
import TestRenderer from 'react-test-renderer'

import PageSidebar from '../PageSidebar'


describe('PageSidebar', () => {

  it('should be a function', () => {
    expect(typeof PageSidebar).toBe('function')
  })

  it('should be a functional component', () => {
    let comp

    comp = TestRenderer.create(
      <PageSidebar/>
    )

    // Functional components don't have instances.
    expect(comp.getInstance()).toBe(null)
  })

  describe('rendered', () => {

    describe('with no props', () => {

      it('should use default props', () => {
        let comp, tree

        comp = TestRenderer.create(
          <PageSidebar/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
      })

    })

    describe('with children', () => {

      it('should render children', () => {
        let comp, tree

        comp = TestRenderer.create(
          <PageSidebar>
            <span>Child</span>
          </PageSidebar>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
      })

    })

    describe('with className', () => {

      it('should render with expected class attribute', () => {
        let comp, tree

        comp = TestRenderer.create(
          <PageSidebar className="Foo"/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
      })

    })

    describe('with compClassName', () => {

      it('should render with expected class attribute', () => {
        let comp, tree

        comp = TestRenderer.create(
          <PageSidebar compClassName=""/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()

        comp = TestRenderer.create(
          <PageSidebar className="foo" compClassName=""/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()

        comp = TestRenderer.create(
          <PageSidebar className="foo" compClassName="page-sidebar"/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
      })

    })

    describe('with compType', () => {

      it('should render with expected component', () => {
        let comp, tree

        comp = TestRenderer.create(
          <PageSidebar compType="aside"/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()

        comp = TestRenderer.create(
          <PageSidebar compType="div"/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()

        comp = TestRenderer.create(
          <PageSidebar compType="section"/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
      })

    })

  })

})
