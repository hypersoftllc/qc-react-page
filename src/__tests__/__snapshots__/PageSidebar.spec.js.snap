// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageSidebar rendered with children should render children 1`] = `
<aside
  className="Block Page-Sidebar"
>
  <span>
    Child
  </span>
</aside>
`;

exports[`PageSidebar rendered with className should render with expected class attribute 1`] = `
<aside
  className="Block Page-Sidebar Foo"
/>
`;

exports[`PageSidebar rendered with compClassName should render with expected class attribute 1`] = `
<aside
  className="Block"
/>
`;

exports[`PageSidebar rendered with compClassName should render with expected class attribute 2`] = `
<aside
  className="Block foo"
/>
`;

exports[`PageSidebar rendered with compClassName should render with expected class attribute 3`] = `
<aside
  className="Block page-sidebar foo"
/>
`;

exports[`PageSidebar rendered with compType should render with expected component 1`] = `
<aside
  className="Block Page-Sidebar"
/>
`;

exports[`PageSidebar rendered with compType should render with expected component 2`] = `
<div
  className="Block Page-Sidebar"
/>
`;

exports[`PageSidebar rendered with compType should render with expected component 3`] = `
<section
  className="Block Page-Sidebar"
/>
`;

exports[`PageSidebar rendered with no props should use default props 1`] = `
<aside
  className="Block Page-Sidebar"
/>
`;
